custom_components/bmw_connected_drive/sensor.py

```diff
@@ -66,21 +120,44 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     account, vehicle, attribute_name, attribute_info
                 )
                 devices.append(device)
+        for service in vehicle.state.attributes:
```

Comment:
Are these always added (not tested your code yet with my non-electric car)? So also for cars that don't support these sensors? We should do a check so that only for car's which have this data the sensors get added.
And can't you combine this into one check so you only have one line where `BMWConnectedDriveSensor` is called?

Reply:
Yep, those services-dicts are always added, even when this is not available for a car.

Reply:
We should not add these sensors in HA for vehicles that don't have this data available.
Please add a check for this.